### PR TITLE
Add registration by role metrics

### PR DESCRIPTION
**Pull request by KaitCrawford**

### CODE REVIEW

**Comment by KaitCrawford** [comment]
@smn ready for review

File: registrations/models.py [hunk 1]
```diff
@@ -251,7 +251,40 @@ def fire_state_metric(sender, instance, created, **kwargs):
 
             total = get_or_incr_cache(
                 total_key,
-                registrations_for_state(state).count)
+                registrations_for_identity_field(
+                    {"details__state": state}).count)
```
Comment (smn) [inline registrations/models.py:255]: Probably a silly question: does this return the `count` or does it return a reference to the [`count()`](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#django.db.models.query.QuerySet.count) function?

**Review by smn — COMMENTED** [review]
It looks good to me but I haven't spent a _huge_ amount of time on this code base. The only thing I have a question about is the `count` reference vs `count()` call.

File: registrations/models.py (hunk 1 above)
Comment (KaitCrawford) [reply registrations/models.py:255]: it returns a reference to the `count()` function
`get_or_incr_cache` takes the key to check and a method to call if the count isn't in the cache

Comment (smn) [reply registrations/models.py:255]: 👍

**Comment by smn** [comment]
:+1: thanks @KaitCrawford

**Pull request merged**